x-pack/plugins/monitoring/server/lib/alerts/fetch_disk_usage_node_stats.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { AlertCluster, AlertDiskUsageNodeStats } from '../../alerts/types';
+
+export async function fetchDiskUsageNodeStats(
+  callCluster: any,
+  clusters: AlertCluster[],
+  index: string,
+  duration: string,
+  size: number
+): Promise<AlertDiskUsageNodeStats[]> {
+  const clustersIds = clusters.map((cluster) => cluster.clusterUuid);
+  const params = {
+    index,
+    filterPath: ['aggregations'],
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                cluster_uuid: clustersIds,
+              },
+            },
+            {
+              term: {
+                type: 'node_stats',
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: `now-${duration}`,
+                },
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
+        clusters: {
+          terms: {
+            field: 'cluster_uuid',
+            size,
+            include: clustersIds,
+          },
+          aggs: {
+            nodes: {
+              terms: {
+                field: 'node_stats.node_id',
+                size,
+              },
+              aggs: {
+                index: {
+                  terms: {
+                    field: '_index',
+                    size: 1,
+                  },
+                },
+                total_in_bytes: {
+                  max: {
+                    field: 'node_stats.fs.total.total_in_bytes',
+                  },
+                },
+                available_in_bytes: {
+                  max: {
+                    field: 'node_stats.fs.total.available_in_bytes',
+                  },
+                },
+                usage_ratio_percentile: {
+                  bucket_script: {
+                    buckets_path: {
+                      available_in_bytes: 'available_in_bytes',
+                      total_in_bytes: 'total_in_bytes',
+                    },
+                    script:
+                      '100 - Math.floor((params.available_in_bytes / params.total_in_bytes) * 100)',
```

Comment:
This seems to calculate free space, rather than used space. I think we want used space here

Reply:
No, this is used space. Notice the `100 - X` in the beginning

Reply:
Oh crap, you're right.

It's confusing because in the panel in the EUI, we show `Disk Available` (which led to my confusion) but you are right - this is correct